test/unit/polygon.js

```diff
@@ -254,4 +254,45 @@
       assert.equal(polygon, null);
     });
   });
+  QUnit.test('_calcDimensions', function(assert) {
+    const polygon = new fabric.Polygon(
+      getPoints(), 
+      { 
+        scaleX: 2, 
+        scaleY: 3,
+        skewX: 20,
+        skewY: 30,
+        strokeWidth: 20,
+        strokeMiterLimit: 10,
+        strokeUniform: false,
+        strokeLineJoin: 'miter',
+        exactBoundingBox: true
+      }
+    ),
+    {
+      left,
+      top,
+      width,
+      height,
+      pathOffset,
+      strokeOffset
+    } = polygon._calcDimensions();
+
+    // Types
+    assert.equal(typeof left, 'number');
+    assert.equal(typeof top, 'number');
+    assert.equal(typeof width, 'number');
+    assert.equal(typeof height, 'number');
+    assert.ok(pathOffset instanceof fabric.Point);
+    assert.ok(strokeOffset instanceof fabric.Point);
+
+    // Values
+    assert.equal(left, 10.485714075442775);
+    assert.equal(top, 14.784917784669414);
+    assert.equal(width, 27.707709196083425);
+    assert.equal(height, 21.750672506349947);
+    assert.deepEqual(pathOffset, new fabric.Point(14.999999999999998, 17.000000000000004));
+    assert.deepEqual(strokeOffset, new fabric.Point(-0.48571407544277534, -2.784917784669414));
```

Comment:
This doesn't make sense, does it? I would expect it to be values > 0

The test is perfect

Reply:
We're calculating the difference when strokeWidth=0 (and no scale, no skew) versus when we could have strokeWidth and scale/skew. In theory it sounds like it should be positive. 

This test has a peculiarity that it is 2 points, and when strokeWidth = 0 I don't really know what happens. I have to think about it.

Reply:
It can happen that there is strokeOffset < 0 when skew is applied. The logic goes something like this:

![image](https://user-images.githubusercontent.com/65685842/198679236-cb6ae308-c45a-4b88-906c-1f2bdf0fb32b.png)

A real case: https://codesandbox.io/s/strokeoffset-mqqxdq

**p2 -> skewY = 0**

`p2._calcDimensions()`

{ strokeOffset: 
  type: "point",
  x: 6.035533905932738,
  y: 2.5
}

**p3 -> skewY = 45**

`p3._calcDimensions()`

{ strokeOffset: 
  type: "point",
  x: 6.035533905932738,
  y: -1.4644660940672622
}

Reply:
I got confused here as well
Thought it was `strokeDiff` that has negative values.
Sorry for the ruckus

Reply:
We should add that to the test as well

Reply:
add here https://github.com/fabricjs/fabric.js/pull/8374/commits/9079fffc54fef8d4066151916f8258f63931afcc